src/NLog/Internal/FileAppenders/BaseFileAppender.cs

```diff
@@ -177,12 +177,12 @@ protected FileStream CreateFileStream(bool allowConcurrentWrite)
                         }
 
                         Directory.CreateDirectory(Path.GetDirectoryName(this.FileName));
-                        return this.TryCreateFileStream(allowConcurrentWrite);
+                        return this.TryCreateFileStream(allowAtomicConcurrentWrite);
                     }
                 }
                 catch (IOException)
                 {
-                    if (!this.CreateFileParameters.ConcurrentWrites || !allowConcurrentWrite || i + 1 == this.CreateFileParameters.ConcurrentWriteAttempts)
```

Comment:
Please add some comments why we should rethrow the exception when `allowConcurrentWrite=false`